libhb/qsv_common.c

```diff
@@ -2177,8 +2175,8 @@ int hb_qsv_full_path_is_enabled(hb_job_t *job)
         return 0;
     }
 
-    // scale_qsv filter can't convert from full to limited range
-    if (job->title->color_range == AVCOL_RANGE_JPEG && job->color_range == AVCOL_RANGE_MPEG)
+    // vpp_qsv filter can't convert from full to limited range, fallback to sw filters until unsupported
+    if (job->title->color_range == AVCOL_RANGE_JPEG)
```

Comment:
Is job color ever set to something that's not AVCOL_RANGE_MPEG?

Reply:
It might be good to change this to check for the case where the title is limited range and the job full range too, in case we add a full range encoding option in the UI.

Reply:
> Is job color_range ever set to something that's not AVCOL_RANGE_MPEG?
Yes, there are AVCOL_RANGE_UNSPECIFIED, AVCOL_RANGE_MPEG, AVCOL_RANGE_JPEG

Reply:
I see that it's hardcoded to AVCOL_RANGE_MPEG in work.c, so if it's accessed before it's giving back the same value as title, argh.
Oh well, another issue to be fixed another time.

Reply:
> It might be good to change this to check for the case where the title is limited range and the job full range too, in case we add a full range encoding option in the UI.

job->color_range is assigned to AVCOL_RANGE_MPEG https://github.com/HandBrake/HandBrake/blob/master/libhb/work.c#L1500
 but this function called before that multiple times and value is AVCOL_RANGE_UNSPECIFIED

By this reason until conversion is not fully supported by ffmpeg vpp_qsv filter is safe to fallback to sw